src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -923,11 +943,43 @@ public partial class JsonStringEnumConverter : System.Text.Json.Serialization.Js
         public sealed override bool CanConvert(System.Type typeToConvert) { throw null; }
         public sealed override System.Text.Json.Serialization.JsonConverter CreateConverter(System.Type typeToConvert, System.Text.Json.JsonSerializerOptions options) { throw null; }
     }
+    [System.Runtime.Versioning.RequiresPreviewFeatures("Polymorphic serialization is currently in preview.")]
+    public enum JsonUnknownDerivedTypeHandling
+    {
+        FailSerialization = 0,
+        FallbackToBaseType = 1,
+        FallbackToNearestAncestor = 2
+    }
     public enum JsonUnknownTypeHandling
     {
         JsonElement = 0,
         JsonNode = 1,
     }
+    [System.Runtime.Versioning.RequiresPreviewFeatures("Polymorphic serialization is currently in preview.")]
+    public partial class JsonPolymorphicTypeConfiguration : System.Collections.Generic.ICollection<(System.Type DerivedType, string? TypeDiscriminatorId)>, System.Collections.Generic.IEnumerable<(System.Type DerivedType, string? TypeDiscriminatorId)>, System.Collections.IEnumerable
+    {
+        public JsonPolymorphicTypeConfiguration(System.Type baseType) { }
+        public System.Type BaseType { get { throw null; } }
```

Comment:
would it make sense to instead have this information directly on JsonTypeInfo? (i.e. `AllowedDerivedTypes`)?

Reply:
Yes. It's a design we could consider once JsonTypeInfoResolvers have been shipped.